src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -0,0 +1,242 @@
+        ///  -------------------------------------------
+        ///  | Id (primary key, varchar) | Data (blob) |
+        ///  -------------------------------------------
+        /// </summary>
+        private const string SolutionDataTableName = "SolutionData" + Version;
+
+        /// <summary>
+        /// Inside the DB we have a table for data that we want associated with a <see cref="Project"/>.
+        /// The data is keyed off of an integral value produced by combining the ID of the Project and
+        /// the ID of the name of the data (see <see cref="SQLitePersistentStorage.ReadStreamAsync(Project, string, CancellationToken)"/>.
+        /// 
+        /// This gives a very efficient integral key, and means that the we only have to store a 
+        /// single mapping from stream name to ID in the string table.
+        /// 
+        /// The format of the table is:
+        /// 
+        ///  ProjectData
+        ///  -------------------------------------------
+        ///  | Id (primary key, integer) | Data (blob) |
+        ///  -------------------------------------------
+        /// </summary>
+        private const string ProjectDataTableName = "ProjectData" + Version;
+
+        /// <summary>
+        /// Inside the DB we have a table for data that we want associated with a <see cref="Document"/>.
+        /// The data is keyed off of an integral value produced by combining the ID of the Document and
+        /// the ID of the name of the data (see <see cref="SQLitePersistentStorage.ReadStreamAsync(Document, string, CancellationToken)"/>.
+        /// 
+        /// This gives a very efficient integral key, and means that the we only have to store a 
+        /// single mapping from stream name to ID in the string table.
+        /// 
+        /// The format of the table is:
+        /// 
+        ///  DocumentData
+        ///  -------------------------------------------
+        ///  | Id (primary key, integer) | Data (blob) |
+        ///  -------------------------------------------
+        /// </summary>
+        private const string DocumentDataTableName = "DocumentData" + Version;
+
+        private const string IdColumnName = "Id";
+        private const string DataColumnName = "Data";
+
+        static SQLitePersistentStorage()
+        {
+            // Attempt to load the correct version of e_sqlite.dll.  That way when we call
+            // into SQLitePCL.Batteries_V2.Init it will be able to find it.
+            var myFolder = Path.GetDirectoryName(
+                typeof(SQLitePersistentStorage).Assembly.Location);
+
+            var is64 = IntPtr.Size == 8;
+            var subfolder = is64 ? "x64" : "x86";
+
+            LoadLibrary(Path.Combine(myFolder, subfolder, "e_sqlite3.dll"));
+
+            // Necessary to initialize SQLitePCL.
+            SQLitePCL.Batteries_V2.Init();
+        }
+
+        [DllImport("kernel32.dll")]
+        private static extern IntPtr LoadLibrary(string dllToLoad);
```

Comment:
Good question! Patching native libraries into the dynamic loader path on Mac and Linux at runtime is tricky. For now, I'd probably say just wrap this with an `if (Environment.OSVersion.Platform == PlatformID.Win32NT) {...}` and make the host responsible for setting it up.